testhelpers/docker_registry.go

```diff
@@ -182,6 +199,14 @@ func (r *DockerRegistry) EncodedLabeledAuth() string {
 	return base64.StdEncoding.EncodeToString([]byte(fmt.Sprintf(`{"username":"%s","password":"%s"}`, r.username, r.password)))
 }
 
+// Add stores the given key name with the provided ImagePrivileges. For example
+// Add("my-image", NewImagePrivileges(Readable, Writable)) will save "my-image" as a
+// readable and writable image into the registry.
+// It must be called before Start to have any effect.
```

Comment:
I don't think it's a good practice to ask the user to remember to call `Add` before calling `Start`.
It might be very confusing to find out the reason that your code isn't running the way you expect it to be because of such thing.
Maybe it's better to pass the `ImagePrivileges` to the `Start` function, or have two `Start` functions:
`StartWithPrivileges` and `StartWithoutPrivileges`.
These are just two ideas. Maybe there are better ones.

By the way, maybe I didn't follow but it looks like you're calling the `Add` function after the `Start` function in your [lifecycle PR](https://github.com/buildpacks/lifecycle/pull/685/files).

Reply:
Whoops, sorry, that was my code suggestion when I didn't understand how it worked!

Reply:
I deleted the comment in the following [commit](https://github.com/buildpacks/imgutil/pull/126/commits/1c2361a45515bb01cc7fe1fae051d620fdc643f3)